rfcs/0185-redistribute-redistributable.md

```diff
@@ -0,0 +1,108 @@
+---
+feature: redistribute-redistributable
+start-date: 2024-12-15
+author: Ekleog
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: https://github.com/NixOS/nixpkgs/issues/83884
+---
+
+# Summary
+[summary]: #summary
+
+Make Hydra build and provide all redistributable software.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Hydra builds only free software and unfree redistributable firmware.
+This means that unfree redistributable software needs to be rebuilt by all the users.
+For example, using MongoDB on a Raspberry Pi 4 (aarch64, which otherwise has access to hydra's cache) takes literally days and huge amounts of swap.
+
+Hydra could provide builds for unfree redistributable software, at minimal added costs.
+This would make life much better for users of such software.
+Especially when the software is still source-available even without being free software, like MongoDB.
+
+# Detailed design
+[design]: #detailed-design
+
+Hydra will build all packages with licenses for which `redistributable = true`.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+With this change, Hydra will start building, among others:
+- CUDA
+- DragonflyDB
+- MongoDB
+- Nomad
+- NVIDIA drivers
+- Outline
+- SurrealDB
+- TeamSpeak
+- Terraform
+- Unrar
```

Comment:
I think `unrar` demonstrates where more care is needed: Unrar has use restrictions, so building it on Hydra adds a legal obligation to Hydra (although in case of unrar it wouldn't be easy to trigger in the infra context)

Reply:
Unrar source you can distribute (and I think binaries too), but you are not allowed to use them to reverse-engineer a rar compressor.

In general, it can be that you can redistribute some binary as is, but if you are running it, you must make sure you are following some restriction correctly.

Reply:
Just now I recalled licensing that disallows "Evil" usage: https://stackoverflow.com/a/18239665

Reply:
It has AFAICT no impact on Hydra? IOW, Hydra redistributing them would not pose any legal risk, as Hydra is definitely not going to reverse-engineer a RAR compressor.

As for the "Evil" usage license, I'd guess it'd fall on the unfree non-redistributable option, but I guess that'd have to be a separate discussion to be had — my main thinking is that if we mark packages as "redistributable" when they can't be redistributed by Hydra, then we have a bug in our packaging.

Reply:
1. You are not just allowing _redistribution_, you are allowing using the package as build input too (otherwise CUDA is pointless)
2. Different redistributable packages have different usage restrictions once you run their code.
3. In the current setup, it is not tracked.

Reply:
Oh right, that's an excellent point! So this most likely requires a new flag like `runnableOnHydra`. I'll add it to the RFC when I get to it, hopefully today!

Reply:
Done :)